plugins/modules/docker_image.py

```diff
@@ -549,9 +557,26 @@ def archive_image(self, name, tag):
             self.log("archive image: image %s not found" % image_name)
             return
 
+        # Will have a 'sha256:' prefix
+        new_image_id = image['Id']
+
+        try:
+            archived_id = archived_image_id(self.archive_path)
+
+            if archived_id:
+                log.debug('Previous image ID was %s', archived_id)
+            else:
+                log.debug('No previous image')
+        except ImageArchiveInvalidException:
+            log.debug('Previous image ID not available', exc_info=True)
+            archived_id = None
+
+        changed = (new_image_id != ('sha256:%s' % archived_id))
+
         self.results['actions'].append('Archived image %s to %s' % (image_name, self.archive_path))
```

Comment:
This only needs to be added if `changed` is `True`. Otherwise, the module is not doing something, and adding this action would then be wrong.

Reply:
Done.